src/Compiler/FSharp.Compiler.Service.fsproj

```diff
@@ -84,6 +84,8 @@
     <InternalsVisibleTo Include="HistoricalBenchmark" />
     <InternalsVisibleTo Include="FSharp.Test.Utilities" />
     <InternalsVisibleTo Include="FSharp.Editor" />
+    <InternalsVisibleTo Include="FsAutoComplete.Core" />
+    <InternalsVisibleTo Include="FsAutoComplete" />
```

Comment:
The FSAC InternalsVisibleTo to allow the cache tweaks

Reply:
This feels very wrong, is this a temporary thing?

Reply:
The transparent compiler itself is marked internal and unless the F# team wants to make it public, this is a better method than forcing FSAC to do reflection all the time.

Reply:
Sure! Here's a more friendly version:

"Yeah, that's true, but there's actually a way to enable it using `FSharpChecker.Create(useTransparentCompiler = true)`. Shouldn't more things be driven by that?

I just think this approach feels like a workaround that avoids proper communication, which seems like a missed opportunity. But if the team doesn't see it that way and everyone's fine with it, then I suppose it's okay.

One other thing that concerns me is that this could potentially be a common request for every FCS user."

Reply:
I wonder if we should before we stabilize it. We can expose necessary parts, but put `Experimental` on them.

Reply:
I don't have a strong preference here. I went with the least invasive changes first to do exactly this, facilitate discussion :)

> One other thing that concerns me is that this could potentially be a common request for every FCS user.

Yeah I agree here. Though, it would be nice to also be able to access other things like [LanguageFeatures](https://github.com/dotnet/fsharp/blob/c9bed7992880912f119baa0124cdb05f3d36d6a2/src/Compiler/Facilities/LanguageFeatures.fsi#L4) instead of [our hack](https://github.com/ionide/FsAutoComplete/blob/fc79e99f59c3da89cca0c1e4367307263228409d/src/FsAutoComplete.Core/FCSPatches.fs#L223)

> I wonder if we should expose everything before we stabilize it? We can expose necessary parts, but put Experimental on them. Or expose them via the checker?

Since FCS makes breaking changes that FSAC needs to handle anyway, it's all kind of the same thing in the end to me. Whatever makes more sense for your team.

Reply:
We'd be happy to have such `InternalsVisibleTo` entries too in Rider plugin. 🙂
We're also used to the changes in APIs.

Reply:
In this case we should just expose everything needed via APIs.